src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -4687,8 +4687,15 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
 	alias->colnames = NIL;
 
 	// get table alias
+	const CMDName *md_alias = table_descr->MdName();
+
+	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
+	{
+		md_alias = table_descr->MdAlias();
+	}
```

Comment:
md_alias is rewritten here and the previous value is never used in this case, so the MdName method is called in vain. What about ternary operator?
```suggestion
	const CMDName *md_alias = 
	    (optimizer_enable_table_alias && NULL != table_descr->MdAlias()) ?
	    table_descr->MdAlias() :
	    table_descr->MdName();
```

Reply:
or
```
const CMDName *md_alias = table_descr->MdAlias();
if (!optimizer_enable_table_alias || NULL == md_alias)
{
	md_alias = table_descr->MdName();
}
```